src/utils/throttle.js

```diff
@@ -0,0 +1,11 @@
+import { throttle as _throttle } from 'lodash'
+
+const throttle = throttleTime => {
```

Comment:
I can use setTimeout with a variable instead this and keep things clear.

Reply:
I don't know I feel like I would reinventing the wheel. I mean not just for this function but for all of them. Soon or later we would use it anyway, wouldn't we? If the size is a problem then we can use some optimization on webpack side. There are already [some solutions](https://github.com/lodash/lodash-webpack-plugin) created for it.

Reply:
Yes, my problem is: 
- optimization (lodash is the biggest bundle size increaser)
- another dependency (lodash already reinvents own JavaScript)

Also, I feel like you can show your code to anybody and if it's pure JS, the person will understand but if it's a zoo of frameworks, it will take much more time.

If you really want to use Lodash, require smaller chunks. You can import only throttle instead all the lib:

```javascript
import throttle from 'lodash/throttle'
```

Although, in the case of context menu, consider `animation-delay` solution. I am not sure it's going to work but if it is, it will save us indeed 😀

Reply:
About the optimization, I agree. If you have it, I do not mind using lodash (in this project).

Reply:
> lodash already reinvents own JavaScript

You made my day. 😄 

I will probably get rid off whole lodash but I am not decided yet. There are still some features that need to be implemented and maybe it will be necessary there. If not it will be gone by the end of my contract.

Reply:
By the way, I am a big fun of decorators but I recently wanted do get rid out of it. I looks weird. If you take a [look at Next project](https://github.com/oacore/next/blob/content/components/accordion/index.jsx#L47-L57), `@bind` is everywhere. It seems to be strange and I do not really like that; bound functions are not really what you always want and it was historically hard to make others familiar with.

I've just looked at [the proposal](https://github.com/tc39/proposal-decorators), it's on stage 2. It takes long and is [explained](https://github.com/tc39/proposal-decorators#why-is-decorators-taking-so-long) as _this causes real problems in the JavaScript ecosystem_.

The final decorators proposal might look differently.

---

Although, I wanted to use MobX, that is [really decorator-based](https://mobx.js.org/refguide/observable-decorator.html) what makes me confused. Despite this there is a way to write different, without decorators syntax and I think this is good approach.

What do you think about decorators?

Reply:
Well, I like them and I think they should be part of our codebase or part of the JS language itself. But I have some rule for it - `not overusing them`. They can save time or make code readable (but they can do the opposite way too). I've started with Python as a first language and that's maybe the reason for my opinion. 😄